internal/freeport/freeport.go

```diff
@@ -0,0 +1,92 @@
+// The MIT License
```

Comment:
There are 2 copies of this across Temporal:
- https://github.com/temporalio/cli/blob/main/temporalcli/devserver/freeport.go
- https://github.com/temporalio/sdk-go/blob/master/testsuite/freeport.go